src/App.tsx

```diff
@@ -27,12 +49,34 @@ class App extends Component<any, AppState> {
 
   generateRandomState() {
     let randNum = Math.floor(Math.random() * 1084);
-    getImage(randNum).then((result) => this.setState({ image: result }));
+    getImage(randNum)
+      .then((result) => {
+        this.setState({ image: result });
+      })
+      .catch((error) => {
+        this.setState({ image: `Error loading image: ${error.toString()}` });
```

Comment:
We will need to figure out button styling or a message saying you cannot save this group